src/main/java/jdraw/std/StdContext.java

```diff
@@ -275,8 +283,13 @@ private void doSave() {
             if (filter instanceof FileNameExtensionFilter && !filter.accept(file)) {
                 file = new File(chooser.getCurrentDirectory(), file.getName() + "." + ((FileNameExtensionFilter) filter).getExtensions()[0]);
             }
-            System.out.println("save current graphic to file " + file.getName() + " using format "
-                    + ((FileNameExtensionFilter) filter).getExtensions()[0]);
+
+
+            try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(file))) {
+                oos.writeObject((Serializable) getModel().getFigures().map(Figure::clone).collect(Collectors.toList()));
```

Comment:
Haven't checked this, i.e. I do not know whether the list used by `Collectors.toList()` is serializable.